iceoryx_posh/source/roudi/port_pool.cpp

```diff
@@ -25,27 +25,27 @@ PortPool::PortPool(PortPoolDataBase& portPoolDataBase) noexcept
 {
 }
 
-cxx::vector<popo::InterfacePortData*, MAX_INTERFACE_NUMBER> PortPool::interfacePortDataList() noexcept
+cxx::list<popo::InterfacePortData, MAX_INTERFACE_NUMBER>& PortPool::interfacePortDataList() noexcept
 {
-    return m_portPoolDataBase->m_interfacePortMembers.content();
+    return m_portPoolDataBase->m_interfacePortMembers;
 }
 
-cxx::vector<popo::ApplicationPortData*, MAX_PROCESS_NUMBER> PortPool::appliactionPortDataList() noexcept
+cxx::list<popo::ApplicationPortData, MAX_PROCESS_NUMBER>& PortPool::appliactionPortDataList() noexcept
 {
-    return m_portPoolDataBase->m_applicationPortMembers.content();
+    return m_portPoolDataBase->m_applicationPortMembers;
 }
 
-cxx::vector<runtime::RunnableData*, MAX_RUNNABLE_NUMBER> PortPool::runnableDataList() noexcept
+cxx::list<runtime::RunnableData, MAX_RUNNABLE_NUMBER>& PortPool::runnableDataList() noexcept
 {
-    return m_portPoolDataBase->m_runnableMembers.content();
+    return m_portPoolDataBase->m_runnableMembers;
 }
 
 cxx::expected<popo::InterfacePortData*, PortPoolError>
 PortPool::addInterfacePort(const std::string& applicationName, const capro::Interfaces interface) noexcept
 {
-    if (m_portPoolDataBase->m_interfacePortMembers.hasFreeSpace())
+    if (!m_portPoolDataBase->m_interfacePortMembers.full())
     {
-        auto interfacePortData = m_portPoolDataBase->m_interfacePortMembers.insert(
+        auto interfacePortData = &m_portPoolDataBase->m_interfacePortMembers.emplace_front(
             iox::cxx::string<100>(iox::cxx::TruncateToCapacity, applicationName), interface);
```

Comment:
No magic numbers. Why `string<100>`?
You could use the alias `using ProcessName_t = cxx::string<100>;` which can be found in `iceoryx_posh_types.hpp`

I know you did not write this particular line of code but the boyscout principle applies and you should always leave the code a little bit cleaner then you found it. 🦘